astropy/io/fits/file.py

```diff
@@ -578,7 +597,7 @@ def _open_zipfile(self, fileobj, mode):
 
         if close:
             zfile.close()
-        self.compression = 'zip'
+        self._file.seek(0)
```

Comment:
First line is moved in the other function, which looks good, but why adding a seek here ? (it's already done in `_open_filename/filelike`)